src/routes/(blog-article)/containerizing-rust-applications-best-practices/+page.md

```diff
@@ -0,0 +1,1049 @@
+```
+
+</CodeBlock>
+
+But you can also create a specific user for your application:
+
+<CodeBlock lang="dockerfile">
+
+```dockerfile
+FROM rust:latest
+
+WORKDIR /app
+
+RUN cargo init
+RUN cargo build --release
+
+RUN groupadd -r appuser
+RUN useradd -r -u 1001 -g appuser appuser
+
+USER appuser
+
+CMD ["./target/release/app"]
+```
+
+</CodeBlock>
+
+Using the `USER` instruction is considered a good practice because by default the container
+will not run as root. In that example, docker will run the container as the user with the ID `1001`.
+
+<Callout type="info">
+
+The Linux Standard Base Core Specification defines three different ranges for
+user IDs, the value from 0 to 99, 100 to 999, and 1000 and above. The first two
+ranges are reserved for system users. All the regular users start from 1000.
+It's also important to note that the user ID 0 is reserved for the root user and
+Linux kernel uses the user id and group id to handle the permissions. The host
+machine and the docker container might have the same user ID but different usernames.
+
+</Callout>
+
+If you run that container, `appuser` will have the same permissions as the user `1001`
+in the host system, whatever it's the name of that user in the host machine.
+
+### Use The `docker run --user` Argument
+
+You can also overwrite the user running the container with the argument `--user`:
+
+<CodeBlock lang="console">
+
+```console
+docker run --user www-data --rm -it docker-rust-app-running-with-root whoami
+www-data
+```
+
+</CodeBlock>
+
+In this example, even though the image is execute as root by default, it will be
+executed as the `www-data` user.
+
+Notice you can even use a non-existing user in both the host and the docker image.
+
+<CodeBlock lang="console">
+
+```console
+$ docker run --user 1001 --rm -it docker-rust-app-running-with-root bash
+I have no name!@895b0f6a3dbb:/app$ 
+```
+
+</CodeBlock>
+
+All this solutions work but they all have a drawback: you need to know the user
+ID at build time (when you build the docker image). You usually want to run the
+container in different environments and sometimes you want to use a different
+user ID for each environment. For example:
+
+- **For development**: If you are using Ubuntu, your user ID is probably `1000`. When
+you run the container locally you want to run it using that ID, so that you don't
+have any problems with permissions.
+- **For CI**: The servers you are using for continuous integration (for instance, GitHub runners)
+might use an specif user. You could use some cache folders and maybe you need to
+use the same user ID as the CI server.
+- **For production**: You could create an specific user for your application and use that
+user ID.
+
+With the proposed solutions you would need to rebuild the docker image so that the
+user ID inside the container is the same as the host user ID.
+
+### Create The User At Runtime
+
+There is al alternative to the previous solutions that make it possible to **run the
```

Comment:
Here it is the opposite.
Here you would write:  There is "an" alternative..... 
Also later in the sentence since it is third person since you are referring to the alternative: ..... that makes it possible....